src/v/utils/unresolved_address.h

```diff
@@ -24,13 +24,17 @@
 /// tuple
 class unresolved_address {
 public:
+    using inet_family = std::optional<ss::net::inet_address::family>;
```

Comment:
This makes sense but why not simply default to ipv4

Reply:
thinking out loud. as a future `fixme:` (unrelated to this tix) - this unresolved addr should really have an ip_address which is already a variant and can contain all the goodies per type.

Reply:
I was trying to repeat the current behavior.  Maybe we can also make resolution optional. For archival it's quite OK to resolve once on startup.